src/snowflake/snowpark/async_job.py

```diff
@@ -206,45 +276,84 @@ def _table_result(
             )
 
     @experimental(version="0.10.0")
-    def result(self) -> Union[List[Row], "pandas.DataFrame", Iterator[Row], int, None]:
+    def result(
+        self,
+        result_type: Optional[
+            Literal["row", "row_iterator", "pandas", "pandas_batches", "none"]
```

Comment:
Do "none" and `None` have the same meaning?

Reply:
they are different, `"none"` means this API will return `None`, and `None` means no result type is provided and defaults to a list of rows. Maybe we should have a better name. Let's use `"no_result"`?